### PR TITLE
Add url handler

### PR DESCRIPTION
**Pull request by mpsijm**
To resolve feedback from our TA, I put the responsibility of parsing/formatting URLs in a new class. 
No functionality has been changed (except for a small regex bug fix).
Have fun reviewing! :D

### CODE REVIEW

File: src/octopeer-github/main/URLHandler.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/**
+ * Created by Maarten on 24-05-2016.
+ *
+ * This utility class handles URL formatting and parsing.
+ */
+class URLHandler {
+
+
```
Comment (thervh70) [inline src/octopeer-github/main/URLHandler.ts]: Double new line?

**Comment by thervh70** [comment]
Nice change!

Merge please.

File: src/octopeer-github/main/URLHandler.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+     *      3: pull request number (in a string)
+     * ]
+     */
+    public static isPullRequestUrl(url: string): Array<string> {
+        const urlFormat = /https?:\/\/.*github\.com\/(.+)\/(.+)\/pull\/([^\/#\?]+)[\/#\?]?.*/;
+        if (urlFormat.test(url)) {
+            const res: RegExpExecArray = urlFormat.exec(url);
+            return [res[0], res[1], res[2], res[3]];
+        } else {
+            return [];
```
Comment (mdingjan) [inline src/octopeer-github/main/URLHandler.ts]: You didn't cover the else statement in your tests. 
Would you like to add a test case for this for 100% coverage for this class? :)

**Comment by mdingjan** [comment]
I've reviewed the changes. I've added one comment for your tests.
When this is fixed, this PR is ready for merging. :)

File: src/octopeer-github/main/URLHandler.ts (hunk 2 above)
Comment (mpsijm) [inline src/octopeer-github/main/URLHandler.ts]: I added it, thanks for noticing! :)

**Comment by mdingjan** [comment]
Nice :) Thanks for processing my feedback. ;)
Will merge!

**Pull request merged**